apollo-federation/src/connectors/json_selection/parser.rs

```diff
@@ -4026,7 +4062,39 @@ mod tests {
         }
     }
 
-    #[cfg(test)]
+    #[test]
+    fn test_disallowed_spread_syntax_error() {
+        assert_eq!(
+            JSONSelection::parse_with_spec("id ...names", ConnectSpec::V0_2),
+            Err(JSONSelectionParseError {
+                message: "nom::error::ErrorKind::Eof".to_string(),
+                fragment: "...names".to_string(),
+                offset: 3,
+                spec: ConnectSpec::V0_2,
+            }),
+        );
+
+        assert_eq!(
+            JSONSelection::parse_with_spec("id ...names", ConnectSpec::V0_3),
+            Err(JSONSelectionParseError {
+                message: "Spread syntax (...) is planned for connect/v0.4".to_string(),
+                // This is the fragment and offset we should get, but we need to
+                // store error offsets in SpanExtra::errors to provide that
+                // information.
+                fragment: "...names".to_string(),
+                offset: 3,
+                spec: ConnectSpec::V0_3,
+            }),
+        );
```

Comment:
The error when you try to use `...` in `ConnectSpec::V0_3` and earlier will be much better than it was in `ConnectSpec::V0_2` (frozen for backwards compatibility).